base/linalg/givens.jl

```diff
@@ -159,7 +159,7 @@ function givensAlgorithm{T<:FloatingPoint}(f::Complex{T}, g::Complex{T})
      # This is a rare case: F is very small.
 
         if f == 0
-            cs = zero
+            cs = zero(T)
```

Comment:
Good catch. As the comment states, it's a rare case, but this might have been slowing down this performance critical function.